rapids_logger/logger.hpp.in

```diff
@@ -0,0 +1,541 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#ifdef @_RAPIDS_LOGGER_MACRO_PREFIX@_BACKWARDS_COMPATIBILITY
+#include <spdlog/spdlog.h>
+#endif
+
+// We only define functions as inline when in backwards compatibility mode
+// because in that mode those functions are included in the header, just
+// defined out of band by the include of logger_impl.hpp into this file. In the
+// new mode the functions are defined in their own TU.
+#ifdef RMM_BACKWARDS_COMPATIBILITY
+#define BACKWARDS_COMPAT_INLINE inline
+#else
+#define BACKWARDS_COMPAT_INLINE
+#endif
+
+#include <memory>
+#include <ostream>
+#include <string>
+
+#ifdef SUPPORTS_LOGGING
+
+// Forward declare spdlog types that are part of our public API.
+namespace spdlog {
+namespace sinks {
+class sink;
+} // namespace sinks
+using sink_ptr = std::shared_ptr<sinks::sink>;
+}  // namespace spdlog
+
+#endif
+
+namespace @_RAPIDS_VISIBILITY_MACRO@ @_RAPIDS_LOGGER_NAMESPACE@ {
+
+namespace detail {
+
+/**
+ * @brief Returns the default log filename for the RMM global logger.
+ *
+ * If the environment variable `RAPIDS_DEBUG_LOG_FILE` is defined, its value is used as the path and
+ * name of the log file. Otherwise, the file `@_RAPIDS_LOGGER_NAMESPACE@_log.txt` in the current working directory is used.
+ *
+ * @return std::string The default log file name.
+ */
+inline std::string default_log_filename()
+{
+  auto* filename = std::getenv("@_RAPIDS_LOGGER_MACRO_PREFIX@_DEBUG_LOG_FILE");
+  // TODO: Do we prefer rmm's default (a file rmm_log.txt) or cudf's default (a
+  // stderr sink)? I think the latter is better.
+  return (filename == nullptr) ? std::string{"@_RAPIDS_LOGGER_NAMESPACE@_log.txt"} : std::string{filename};
+}
+
+/**
+ * @brief Represent a size in number of bytes.
+ */
+struct bytes {
+  std::size_t value;  ///< The size in bytes
+
+#ifdef RMM_BACKWARDS_COMPATIBILITY
```

Comment:
You're right that strictly speaking it doesn't, but the only reason that this was previously defined is so that you can pass these objects directly to an fmt formatter. We could leave it unconditionally defined too if we want, that would be fine.

Reply:
No longer relevant, backwards compat layers removed.